component/src/main/java/io/siddhi/extension/io/file/listeners/FileCronListener.java

```diff
@@ -0,0 +1,151 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.siddhi.extension.io.file.listeners;
+
+import io.siddhi.core.stream.input.source.SourceEventListener;
+import io.siddhi.extension.io.file.processors.FileProcessor;
+import io.siddhi.extension.io.file.util.Constants;
+import io.siddhi.extension.io.file.util.FileSourceConfiguration;
+import io.siddhi.extension.io.file.util.VFSClientConnectorCallback;
+import org.apache.log4j.Logger;
+import org.quartz.CronScheduleBuilder;
+import org.quartz.Job;
+import org.quartz.JobBuilder;
+import org.quartz.JobDataMap;
+import org.quartz.JobDetail;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.JobKey;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.Trigger;
+import org.quartz.TriggerBuilder;
+import org.quartz.impl.StdSchedulerFactory;
+import org.wso2.carbon.messaging.exceptions.ClientConnectorException;
+import org.wso2.transport.file.connector.sender.VFSClientConnector;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * FileCron Listener is executted when the cron expression is given. If the current time satisfied by the cron
+ * expression then the file processing will be executed.
+ */
+public class FileCronListener implements Job {
+    private static final Logger log = Logger.getLogger(FileSystemListener.class);
+    SourceEventListener sourceEventListener;
+
+    public FileCronListener() {
+    }
+
+    public static void scheduleJob(FileSourceConfiguration fileSourceConfiguration, FileProcessor fileProcessor,
+                                   VFSClientConnector vfsClientConnector) {
+        try {
+            JobKey jobKey = new JobKey("job", "group");
+
+            Scheduler scheduler = new StdSchedulerFactory().getScheduler();
+            fileSourceConfiguration.setScheduler(scheduler);
+            if (scheduler.checkExists(jobKey)) {
+                scheduler.deleteJob(jobKey);
+            }
+
+            scheduler.start();
+            JobDataMap dataMap = new JobDataMap();
+            dataMap.put(Constants.FILE_SOURCE_CONFIGURATION, fileSourceConfiguration);
+            dataMap.put(Constants.FILE_PROCESSOR, fileProcessor);
+            dataMap.put(Constants.VFS_CLIENT_CONNECTOR, vfsClientConnector);
+
+            JobDetail cron = JobBuilder.newJob(FileCronListener.class)
+                    .usingJobData(dataMap)
+                    .withIdentity(jobKey)
+                    .build();
+
+            Trigger trigger = TriggerBuilder
+                    .newTrigger()
+                    .withIdentity("TriggerName", "group")
```

Comment:
Lets make those constants

Reply:
fixed in 47cc199b8188f3701d228a031aeb7a33f02bc856